tests/Rules.Framework.Tests/Extensions/RulesEngineExtensionsTests.cs

```diff
@@ -0,0 +1,32 @@
+namespace Rules.Framework.Tests.Extensions
+{
+    using FluentAssertions;
+    using Moq;
+    using Rules.Framework.Evaluation;
+    using Rules.Framework.Extension;
+    using Rules.Framework.Tests.TestStubs;
+    using Rules.Framework.Validation;
+    using Xunit;
+
+    public class RulesEngineExtensionsTests
+    {
+        [Fact]
+        public void RulesEngineExtensions_ToGenericEngine_Success()
+        {
+            // Arrange
+            var rulesEngine = new RulesEngine<ContentType, ConditionType>(
+                Mock.Of<IConditionsEvalEngine<ConditionType>>(),
+                Mock.Of<IRulesDataSource<ContentType, ConditionType>>(),
+                Mock.Of<IValidatorProvider>(),
+                RulesEngineOptions.NewWithDefaults(),
+                Mock.Of<IConditionTypeExtractor<ContentType, ConditionType>>()
+                );
+
+            //Act
+            var genericEngine = rulesEngine.CreateGenericEngine();
+
+            //Arrange
+            genericEngine.Should().NotBeNull();
```

Comment:
Perhaps add also assert to ensure `genericEngine` is also of type `GenericRulesEngine`?

Reply:
Done